src/seedu/addressbook/Main.java

```diff
@@ -107,6 +108,7 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            storage.checkStorageFileExists();
```

Comment:
The exception should be thrown from StorageFile::save() method.
The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.

if not you can catch the specific exception and warn the user of the error and continue.